template/src/ontology/Makefile.jinja2

```diff
@@ -393,233 +425,231 @@ no-mirror-refresh-%:
 # ----------------------------------------
 
-# Test patterns for schema compliance:
-
-.PHONY: patterns
-patterns: all_imports $(PATTERNDIR)/pattern.owl $(PATTERNDIR)/definitions.owl
+PAT=true # Global parameter to bypass pattern generation
+ALL_PATTERN_FILES=$(wildcard $(PATTERNDIR)/dosdp-patterns/*.yaml)
+ALL_PATTERN_NAMES=$(strip $(patsubst %.yaml,%, $(notdir $(wildcard $(PATTERNDIR)/dosdp-patterns/*.yaml))))
 
 .PHONY: pattern_clean
 pattern_clean:
-	echo "Not implemented"
+	rm -f $(DOSDP_OWL_FILES_DEFAULT)
+	rm -f $(DOSDP_TERM_FILES_DEFAULT){% if project.pattern_pipelines_group is defined %}
+	{%- for pipeline in project.pattern_pipelines_group.products %}
+	rm -f $(DOSDP_OWL_FILES_{{ pipeline.id.upper() }})
+	rm -f $(DOSDP_TERM_FILES_{{ pipeline.id.upper() }})
+	{%- endfor %}{% endif %}
+
+.PHONY: patterns
+patterns dosdp:
+	echo "Validating all DOSDP templates"
+	make dosdp_validation
+	echo "Building $(PATTERNDIR)/definitions.owl"
+	make $(PATTERNDIR)/pattern.owl $(PATTERNDIR)/definitions.owl
+
+# DOSDP Template Validation
+
+$(TMPDIR)/pattern_schema_checks: $(ALL_PATTERN_FILES) | $(TMPDIR)
+	$(PATTERN_TESTER) $(PATTERNDIR)/dosdp-patterns/ && touch $@
 
 .PHONY: pattern_schema_checks
-pattern_schema_checks: update_patterns
-	$(PATTERN_TESTER) $(PATTERNDIR)/dosdp-patterns/
+pattern_schema_checks dosdp_validation: $(TMPDIR)/pattern_schema_checks
 
-#This command is a workaround for the absence of -N and -i in wget of alpine (the one ODK depend on now). It downloads all patterns specified in external.txt
 .PHONY: update_patterns
-update_patterns: .FORCE
-	if [ $(PAT) = true ]; then rm -f $(PATTERNDIR)/dosdp-patterns/*.yaml.1 || true; fi
-	if [ $(PAT) = true ] && [ -s $(PATTERNDIR)/dosdp-patterns/external.txt ]; then wget -i $(PATTERNDIR)/dosdp-patterns/external.txt --backups=1 -P $(PATTERNDIR)/dosdp-patterns; fi
-	if [ $(PAT) = true ]; then rm -f $(PATTERNDIR)/dosdp-patterns/*.yaml.1 || true; fi
+update_patterns: download_patterns
+	cp -r $(TMPDIR)/dosdp/*.yaml $(PATTERNDIR)/dosdp-patterns
 
+# This command is a workaround for the absence of -N and -i in wget of alpine (the one ODK depend on now).
+# It downloads all patterns specified in external.txt
+.PHONY: download_patterns
+download_patterns:
+	if [ $(PAT) = true ]; then rm -f $(TMPDIR)/dosdp/*.yaml.1 || true; fi
+	if [ $(PAT) = true ] && [ -s $(PATTERNDIR)/dosdp-patterns/external.txt ]; then wget -i $(PATTERNDIR)/dosdp-patterns/external.txt --backups=1 -P $(TMPDIR)/dosdp; fi
+	if [ $(PAT) = true ]; then rm -f $(TMPDIR)/dosdp/*.yaml.1 || true; fi
 
-$(PATTERNDIR)/pattern.owl: pattern_schema_checks update_patterns
-	if [ $(PAT) = true ]; then $(DOSDPT) prototype --obo-prefixes true --template=$(PATTERNDIR)/dosdp-patterns --outfile=$@; fi
+$(PATTERNDIR)/dospd-patterns/%.yml: download_patterns
+	cmp -s $(TMPDIR)/dosdp-$*.yml $@; RETVAL=$$?; if [ $$RETVAL -eq 0 ]; then echo "Mirror identical, ignoring."; else echo "Mirrors different, updating." && mv $(TMPDIR)/dosdp/$*.yml $@; fi
```

Comment:
Same as above: missing test on the existence of the file before attempting the comparison, and unnecessary use of `RETVAL`.